show/main.py

```diff
@@ -129,6 +176,156 @@ def run_command(command, display_cmd=False):
+    if mode is None:
+        mode = "default"
+    return mode
+
+
+# Global class instance for SONiC interface name to alias conversion
+iface_alias_converter = InterfaceAliasConverter()
+
+
+def print_output_in_alias_mode(output, index):
+    """Convert and print all instances of SONiC interface
+       name to vendor-sepecific interface aliases.
+    """
+
+    alias_name = ""
+    interface_name = ""
+
+    # Adjust tabulation width to length of alias name
+    if output.startswith("---"):
+        word = output.split()
+        dword = word[index]
+        underline = dword.rjust(iface_alias_converter.alias_max_length,
+                                '-')
+        word[index] = underline
+        output = '  ' .join(word)
+
+    # Replace SONiC interface name with vendor alias
+    word = output.split()
+    if word:
+        interface_name = word[index]
+        interface_name = interface_name.replace(':', '')
+    for port_name in natsorted(iface_alias_converter.port_dict.keys()):
+            if interface_name == port_name:
+                alias_name = iface_alias_converter.port_dict[port_name]['alias']
+    if alias_name:
+        if len(alias_name) < iface_alias_converter.alias_max_length:
+            alias_name = alias_name.rjust(
+                                iface_alias_converter.alias_max_length)
+        output = output.replace(interface_name, alias_name, 1)
+
+    click.echo(output.rstrip('\n'))
+
+
+def run_command_in_alias_mode(command):
+    """Run command and replace all instances of SONiC interface names
+       in output with vendor-sepecific interface aliases.
+    """
+
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+
+    while True:
+        output = process.stdout.readline()
+        if output == '' and process.poll() is not None:
+            break
+
+        if output:
+            index = 1
+            raw_output = output
+            output = output.lstrip()
+
+            if command.startswith("portstat"):
+                """Show interface counters"""
+                index = 0
+                if output.startswith("IFACE"):
+                    output = output.replace("IFACE", "IFACE".rjust(
+                               iface_alias_converter.alias_max_length))
+                print_output_in_alias_mode(output, index)
+
+            elif command == "pfcstat":
+                """Show pfc counters"""
+                index = 0
+                if output.startswith("Port Tx"):
+                    output = output.replace("Port Tx", "Port Tx".rjust(
+                                iface_alias_converter.alias_max_length))
+
+                elif output.startswith("Port Rx"):
+                    output = output.replace("Port Rx", "Port Rx".rjust(
+                                iface_alias_converter.alias_max_length))
+                print_output_in_alias_mode(output, index)
+
+            elif command == "sudo sfputil show eeprom":
```

Comment:
This should be `command.startswith("sudo sfputil show eeprom")` because the command can take flags and an optional interface argument.

You also need to add code to translate the optional interface name from vendor alias to SONiC name in the "show interfaces transceiver ..." commands on lines 506-548 like you did in the other commands (see lines 560-562 for reference).

Reply:
Addressed and updated in new commit.. 👍